### PR TITLE
Header donation

### PR DESCRIPTION
**Pull request by duranmla**
**What:**
Add support to inject custom donation component

**Why:**
relates to the effort of
[Gatsby link for Donate button](https://app.asana.com/0/1168997577035609/1198926861866063)

**How:**
Use the `slot` API from stencil to allow render a custom component whenever the `donateurl` is set to falsy value
_The need of use a donateurl to false comes from the bug that has been addressed with https://github.com/ionic-team/stencil/pull/2650_

**Extras:**
This code will add donate button on mobile and desktop for users and guests, besides some other aesthetics details.

**Media:**
- [Aesthetics updates](https://www.loom.com/share/ae8c37f4e96c4065bb1e9e504e46ae62)
- [How the slot API works with our component](https://www.loom.com/share/ee4d613691fe41feb14e5a345b71a734)

### CODE REVIEW

File: packages/header-component/src/components/dc-header/dc-header.tsx [hunk 1]
```diff
@@ -205,7 +205,16 @@ export class Header {
           </nav>
           <div class="session-items">
             {this.user ? (
-              <dc-user-items user={this.user} community={this.community} />
+              <div class="with-user">
+                <slot name="donate">
+                  {this.donateurl && (
```
Comment (orlando) [inline packages/header-component/src/components/dc-header/dc-header.tsx:210]: Can we remove this now the fix has landed on Stencil?

**Review by orlando — COMMENTED** [review]
I left a comment related to the workaround we are doing

**Review by arathjz — APPROVED** [review]

File: packages/header-component/src/components/dc-header/dc-header.tsx (hunk 1 above)
Comment (duranmla) [reply packages/header-component/src/components/dc-header/dc-header.tsx:210]: It has been merge to master on the stenciljs project but it hasn't get released yet @orlando 
https://github.com/ionic-team/stencil/commits/master currently the latest version is 2.3.0 and it doesn't have the fix yet.

Comment (duranmla) [reply packages/header-component/src/components/dc-header/dc-header.tsx:210]: I have created a [technical debt](https://app.asana.com/0/1195530778765030/1199244186304413) for this one so we can move forward now

**Pull request merged**